python/cudf/cudf/core/udf/strings_lowering.py

```diff
@@ -193,6 +230,20 @@ def call_create_string_view_from_udf_string(udf_str, sv):
     _create_string_view_from_udf_string(udf_str, sv)
 
 
+# Return string setitem impl with an extra incref
+@cuda_lower(
+    operator.setitem,
+    types.CPointer(managed_udf_string),
+    types.Integer,
+    types.Any,
+)
+def setitem_cpointer_managed_udf_string(context, builder, sig, args):
+    base_ptr, idx, val = args
+    elem_ptr = builder.gep(base_ptr, [idx])
+    builder.store(val, elem_ptr)
```

Comment:
Following on from the discussion just below - I think this note generically explains why it makes sense to incref when we store a Managed UDF String in a CPointer array (without requiring the reader to think about the big picture of what happens when the kernel exits):

```suggestion
    builder.store(val, elem_ptr)
    # Storing a Managed UDF String in a CPointer array effectively creates a
    # new reference; represent this by incrementing the refcount of the source
    # of the assignment
```